File: C, C++/quick sort.c
#include<stdio.h>
int partition(int a[],int low, int high)
{
    int i=low;
    int j=high+1;
    int c,pivot;
    pivot=a[low];

 while(j>i){
        do{
            i++;
        }while(a[i]<pivot);
        do
        {
            j--;
        }while(a[j]>pivot);
        if(i<j)
        {
            c=a[i];
            a[i]=a[j];
            a[j]=c;
        }



}   if (i>j){
    c=a[low];
    a[low]=a[j];
    a[j]=c;}

    return j;
}
void quicksort(int a[],int low,int high)
{
    if(low<high)
    {
        int j=partition(a,low,high);
        quicksort(a,low,j-1);
        quicksort(a,j+1,high);
    }

}
void printarray(int a[],int n)
{
    int i;
    for(i=0;i<n;i++)
    {
        printf("%d ",a[i]);
    }
}



void main()
{

    int a[]={8,7,28,1,0,6,9,10,3};
    int n=sizeof(a)/sizeof(a[0]);
    quicksort(a,0,n-1);
    printarray(a,n);



}

File: Java/quick_sort.java
import java.util.*;
class quick_sort
{ 
    
    int partition(int arr[], int low, int high) 
    { 
        int pivot = arr[high];  
        int i = (low-1); 
        for (int j=low; j<high; j++) 
        { 
           
            if (arr[j] < pivot) 
            { 
                i++; 
                int temp = arr[i]; 
                arr[i] = arr[j]; 
                arr[j] = temp; 
            } 
        } 
  
        
        int temp = arr[i+1]; 
        arr[i+1] = arr[high]; 
        arr[high] = temp; 
  
        return i+1; 
    } 
  
    void sort(int arr[], int low, int high) 
    { 
        if (low < high) 
        { 
            
            int pi = partition(arr, low, high); 
            sort(arr, low, pi-1); 
            sort(arr, pi+1, high); 
        } 
    } 
  
    static void printArray(int arr[]) 
    { 
        int n = arr.length; 
        for (int i=0; i<n; ++i) 
            System.out.print(arr[i]+" "); 
        System.out.println(); 
    }
     
    public static void main(String args[]) 
    { 
        quick_sort ob = new quick_sort();
        
	Scanner sc=new Scanner(System.in);
	
	System.out.println("enter the length of array");
	int n=sc.nextInt();
        int arr[]=new int[n];
	System.out.println();
	System.out.println("enter the elements");
	for(int i=0;i<n;i++)
	{
	     arr[i]=sc.nextInt();
	} 
         
	System.out.println("\nUnsorted array");
	printArray(arr);
        
	ob.sort(arr, 0, n-1); 
        
	System.out.println("\nSorted array"); 
        printArray(arr);
    } 
} 
